src/components/ProductActions/Pull.js

```diff
@@ -0,0 +1,174 @@
+import { useParams } from 'react-router-dom'
+import Axios from 'axios'
+import target from 'api/api.target'
+import { useProducts } from "contexts/ProductsContext";
+import { useNavigation } from 'contexts/NavigationContext'
+
+const useStyles = makeStyles(theme=>({
+    img: {
+        display: "block",
+        maxWidth: "70%",
+        border: "1px solid lightgray",
+        borderRadius: "5%",
+        margin: "0 auto",
+    },
+    container: {
+        margin: "0 auto",
+        maxWidth: "80%",
+        [theme.breakpoints.up("md")]:{
+            maxWidth: "60%"
+        },
+        [theme.breakpoints.up("lg")]:{
+            maxWidth: "40%"
+        }
+    },
+    greenBox: {
+        backgroundColor: "#dbfad4",
+        marginTop: "1rem",
+    },
+    txt: {
+        fontSize: "1.1rem",
+        margin: "5px",
+        textAlign: "center"
+    }
+}))
+
+function Pull(props) {
+    const { product_id, storage_id} = useParams()
+    const [product, setProduct] = React.useState(null);
+    const [input, setInput] = React.useState(0);
+    const [productStorageDetails, setProductStorageDetails] = React.useState(null);
+    const [submitting, setSubmitting] = React.useState(false)
+    const [open, setOpen] = React.useState(false)
+    const [response, setResponse] = React.useState('')
+    const { getProductDetails } = useProducts();
+    const { goBack } = useNavigation();
+
+    async function getProduct() {
+        let product = await getProductDetails(product_id);
+        setProduct(product);
+        // console.log(product);
+    }
+    
+    const getProductStorageDetails = () => {
+        // console.log(`${target}/getThisProductInAllStorages/${product_id}`)
+        Axios.get(`${target}/storages/getThisProductInAllStorages/${product_id}`)
+        .then(res => {
+            const sto_id = storage_id
+            const result = res.data.find(({storage_id}) =>storage_id === parseInt(sto_id))
+            setProductStorageDetails(result)
+            // console.log(result)
+        })
+        .catch(err=>console.log(err))
+    }
+
+    const submit = () => {
+        if (input <= productStorageDetails.quantity && input > 0){
+            setSubmitting(true)
+            // console.log(input)
+            // setTimeout(()=>{
+                // setResponse('OK!')
+                // setOpen(true)
+                // setSubmitting(false)
+            // }, 3000)
+            Axios.patch(`${target}/storages/pull`, {product_id, storage_id, quantity: input})
+                .then(res=>{
+                    setResponse(res.data)
+                    setOpen(true)
+                    setSubmitting(false)
+                    //Expensive design choice right here
+                    getProductStorageDetails();
+                })
+        }
+        else if (input > productStorageDetails.quantity) {
+            setResponse('You are pulling way too much!')
```

Comment:
Not a big one, but grammatically should be "many" as items are countable nouns like people, clothes, etc. As compared to "much" for uncountable nouns like sugar, water. People always corrected me with this one hahaha so I'm just passing the favor

Reply:
I like these grammar lessons from victa senpai ❤️